lib/src/value/color.ts

```diff
@@ -3,202 +3,299 @@
 }
 
-interface HwbColor {
-  hue: number;
-  whiteness: number;
-  blackness: number;
-  alpha?: number;
+function emitColor4ApiDeprecation(name: string) {
+  console.warn(`\`${name}\` is deprecated, use \`channel\` instead.`);
 }
 
 /** A SassScript color. */
 export class SassColor extends Value {
-  private redInternal?: number;
-  private greenInternal?: number;
-  private blueInternal?: number;
-  private hueInternal?: number;
-  private saturationInternal?: number;
-  private lightnessInternal?: number;
-  private readonly alphaInternal: number;
-
-  constructor(color: RgbColor);
-  constructor(color: HslColor);
-  constructor(color: HwbColor);
-  constructor(color: RgbColor | HslColor | HwbColor) {
+  private color: PlainColorObject;
+
+  constructor(options: Channels & {space?: KnownColorSpace}) {
     super();
 
-    if ('red' in color) {
-      this.redInternal = fuzzyAssertInRange(
-        Math.round(color.red),
-        0,
-        255,
-        'red'
-      );
-      this.greenInternal = fuzzyAssertInRange(
-        Math.round(color.green),
-        0,
-        255,
-        'green'
-      );
-      this.blueInternal = fuzzyAssertInRange(
-        Math.round(color.blue),
-        0,
-        255,
-        'blue'
-      );
-    } else if ('saturation' in color) {
-      this.hueInternal = positiveMod(color.hue, 360);
-      this.saturationInternal = fuzzyAssertInRange(
-        color.saturation,
-        0,
-        100,
-        'saturation'
-      );
-      this.lightnessInternal = fuzzyAssertInRange(
-        color.lightness,
-        0,
-        100,
-        'lightness'
-      );
-    } else {
-      // From https://www.w3.org/TR/css-color-4/#hwb-to-rgb
-      const scaledHue = positiveMod(color.hue, 360) / 360;
-      let scaledWhiteness =
-        fuzzyAssertInRange(color.whiteness, 0, 100, 'whiteness') / 100;
-      let scaledBlackness =
-        fuzzyAssertInRange(color.blackness, 0, 100, 'blackness') / 100;
-
-      const sum = scaledWhiteness + scaledBlackness;
-      if (sum > 1) {
-        scaledWhiteness /= sum;
-        scaledBlackness /= sum;
-      }
-
-      // Because HWB is (currently) used much less frequently than HSL or RGB, we
-      // don't cache its values because we expect the memory overhead of doing so
-      // to outweigh the cost of recalculating it on access. Instead, we eagerly
-      // convert it to RGB and then convert back if necessary.
-      this.redInternal = hwbToRgb(
-        scaledHue + 1 / 3,
-        scaledWhiteness,
-        scaledBlackness
-      );
-      this.greenInternal = hwbToRgb(
-        scaledHue,
-        scaledWhiteness,
-        scaledBlackness
-      );
-      this.blueInternal = hwbToRgb(
-        scaledHue - 1 / 3,
-        scaledWhiteness,
-        scaledBlackness
+    if (options.alpha === null && !options.space) {
+      console.warn(
+        'Passing `alpha: null` without setting `space` is deprecated.\n\nMore info: https://sass-lang.com/d/null-alpha'
```

Comment:
What's the correct way to handle deprecations?

Reply:
I think `console.warn()` is probably fine for the embedded host. That's fairly standard for JS libraries, right?